src/components/asset-list/RecyclerAssetList2/Claimable.tsx

```diff
@@ -1,73 +1,37 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { Box, Inline, Stack, Text } from '@/design-system';
 import { useAccountSettings } from '@/hooks';
-import { useClaimables } from '@/resources/addys/claimables/query';
+import { useClaimablesStore } from '@/state/claimables/claimables';
 import { FasterImageView } from '@candlefinance/faster-image';
 import { ButtonPressAnimation } from '@/components/animations';
 import { deviceUtils } from '@/utils';
 import Routes from '@/navigation/routesNames';
 import { ExtendedState } from './core/RawRecyclerList';
-import { convertAmountAndPriceToNativeDisplay, convertAmountToNativeDisplayWorklet, convertRawAmountToBalance } from '@/helpers/utilities';
+import { convertAmountToNativeDisplayWorklet } from '@/helpers/utilities';
 import { analyticsV2 } from '@/analytics';
 import { ChainImage } from '@/components/coin-icon/ChainImage';
-import { useNativeAsset } from '@/utils/ethereumUtils';
 import { ChainId } from '@/state/backendNetworks/types';
-import { usePoints } from '@/resources/points';
 
 const RAINBOW_ICON_URL = 'https://rainbowme-res.cloudinary.com/image/upload/v1694722625/dapps/rainbow-icon-large.png';
 
 export const Claimable = React.memo(function Claimable({ uniqueId, extendedState }: { uniqueId: string; extendedState: ExtendedState }) {
-  const { accountAddress, nativeCurrency } = useAccountSettings();
+  const { nativeCurrency } = useAccountSettings();
   const { navigate } = extendedState;
 
   const isETHRewards = uniqueId === 'rainbow-eth-rewards';
 
-  const eth = useNativeAsset({ chainId: ChainId.mainnet });
+  const claimable = useClaimablesStore(state => state.getData())?.claimables.find(claimable => claimable.uniqueId === uniqueId);
```

Comment:
Considering the update patterns (all claimables update as a block after each fetch) and the fact that the full claimables data is already accessed higher up in the list, it’d likely be quite a bit faster to pass down the items directly. That way you skip both the O(n) lookups in each list item, plus the selector overhead. Same likely goes for positions and any other list data that matches this pattern.